plugins/analogix/fu-analogix-common.c

```diff
@@ -0,0 +1,231 @@
+/*
+ * Copyright (C) 2021 Xiaotian Cui <xtcui@analogixsemi.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+#include "fu-analogix-common.h"
+
+guint64
+hex_str_to_dec (const gchar *str, guint8 len)
+{
+	g_autofree gchar *src_str = NULL;
+
+	if (len > 8 || str == NULL)
+		return 0;
+	src_str = g_strndup (str, len);
+
+	return g_ascii_strtoull (src_str, NULL, 16);
+}
+
+gboolean
+parse_fw_hex_file (const guint8 *fw_src, guint32 src_fw_size,
+		   AnxImgHeader *out_header, guint8 *out_binary)
+{

```

Comment:
this function is only used by the `fu-analogix-firmware.c` as I can tell.  should it be moved there and then made into a static function?

Reply:
I think the plan needs to be "_subclass FuIhexFirmware and use the tokenization there_"